src/include/session.h

```diff
@@ -30,6 +30,28 @@ struct __wt_hazard {
 #endif
 };
 
+/*
+ * WT_HAZARD_WEAK --
+ *	A weak hazard pointer.
+ */
+struct __wt_hazard_weak {
+    WT_REF *ref; /* Page reference */
+    bool valid;  /* Is the weak hazard pointer still valid? */
+};
+
+/*
+ * WT_HAZARD_WEAK_ARRAY --
```

Comment:
Shouldn't we have `WT_HAZARD_ARRAY` and have the `__wt_session_impl` declare two forms of it, i.e.:
```
     WT_HAZARD_ARRAY *hazard_weak;
     WT_HAZARD_ARRAY *hazard;
```
I guess the slight difference in types between weak and regular hazard pointers may make this hard. But it does seem weird to me that we have a proper array type on the session and then the original hazard array.

Reply:
I think we should consider a different name. `WT_HAZARD_WEAK_ARRAY` is actually a struct that contains part of the weak array along with several other items.

`WT_HAZARD_WEAK_NODE`?

Reply:
The node represents the latest active array where weak pointers will be set. The other arrays in the list are only going to be cleared as transactions resolve. So, the rest of the nodes are kind of inactive.

This is where some code refactoring will help us eventually. The new array type could become a generic container for arrays, and the active hazards could use them as well. The active could grow the array inside the container and the weak could attach new containers of larger array size to create a list.

We should do something like that when we have more confidence in the code.

For now how about `WT_HAZARD_WEAK_CONTAINER` , I am okay with `WT_HAZARD_WEAK_NODE` as well.

Reply:
I am okay to leave it as-is for now and think about renaming when we are ready to refactor.

Reply:
Marking as resolved - leaving for a later refactoring.